drivers/display/display_gc9a01a.c

```diff
@@ -0,0 +1,599 @@
+	return gc9a01a_transmit(dev, GC9A01A_DISPON, NULL, 0);
+}
+
+/**
+ * @brief To enter into DISPLAY OFF mode
+ *
+ * @param dev GC9A01A device instance.
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_display_blanking_on(const struct device *dev)
+{
+	LOG_DBG("Turning display blanking on");
+	return gc9a01a_transmit(dev, GC9A01A_DISPOFF, NULL, 0);
+}
+
+/**
+ * @brief To set the pixel format
+ *
+ * @param dev GC9A01A device instance.
+ * @param pixel_format Display pixel format
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pixel_format)
+{
+	struct gc9a01a_data *data = dev->data;
+
+	int r;
+	uint8_t tx_data;
+	uint8_t bytes_per_pixel;
+
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		bytes_per_pixel = 2U;
+		tx_data = GC9A01A_PIXSET_MCU_16_BIT | GC9A01A_PIXSET_RGB_16_BIT;
+	} else if (pixel_format == PIXEL_FORMAT_RGB_888) {
+		bytes_per_pixel = 3U;
+		tx_data = GC9A01A_PIXSET_MCU_18_BIT | GC9A01A_PIXSET_RGB_18_BIT;
+	} else {
+		LOG_ERR("Unsupported pixel format");
+		return -ENOTSUP;
+	}
+
+	r = gc9a01a_transmit(dev, GC9A01A_PIXFMT, &tx_data, 1U);
+	if (r < 0) {
+		return r;
+	}
+
+	data->pixel_format = pixel_format;
+	data->bytes_per_pixel = bytes_per_pixel;
+
+	return 0;
+}
+
+/**
+ * @brief To set the display orientation.
+ *
+ * @param dev GC9A01A device instance.
+ * @param orientation Display orientation
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_set_orientation(const struct device *dev,
+				   const enum display_orientation orientation)
+{
+	struct gc9a01a_data *data = dev->data;
+
+	int r;
+	uint8_t tx_data = GC9A01A_MADCTL_BGR;
+
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) { // works 0
+		tx_data |= 0;
+	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_90) { // works CW 90
+		tx_data |= GC9A01A_MADCTL_MV | GC9A01A_MADCTL_MY;
+	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_180) { // works CW 180
+		tx_data |= GC9A01A_MADCTL_MY | GC9A01A_MADCTL_MX | GC9A01A_MADCTL_MH;
+	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_270) { // works CW 270
+		tx_data |= GC9A01A_MADCTL_MV | GC9A01A_MADCTL_MX;
+	}
+
+	r = gc9a01a_transmit(dev, GC9A01A_MADCTL, &tx_data, 1U);
+	if (r < 0) {
+		return r;
+	}
+
+	data->orientation = orientation;
+
+	return 0;
+}
+
+/**
+ * @brief To do overall display device configuration.
+ *
+ * @param dev GC9A01A device instance.
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_configure(const struct device *dev)
+{
+	const struct gc9a01a_config *config = dev->config;
+
+	int r;
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+
+	/* pixel format */
+	if (config->pixel_format == GC9A01A_PIXEL_FORMAT_RGB565) {
+		pixel_format = PIXEL_FORMAT_RGB_565;
+	} else {
+		pixel_format = PIXEL_FORMAT_RGB_888;
+	}
+
+	r = gc9a01a_set_pixel_format(dev, pixel_format); // Set pixel format.
```

Comment:
Why not pass the cfg pixel format directly here and prevent bad values via `BUILD_CHECK`?